src/MiniCover/Commands/Options/CoverageLoadedFileOption.cs

```diff
@@ -0,0 +1,36 @@
+﻿using MiniCover.Model;
+using Newtonsoft.Json;
+using System;
+using System.IO;
+
+namespace MiniCover.Commands.Options
+{
+    internal class CoverageLoadedFileOption : CoverageFileOption, IMiniCoverOption<InstrumentationResult>
```

Comment:
This inheritance doesn't look very good.
It has some hacks to override the value type.
Maybe you can create remove this Option and add a CoverageLoadedBaseCommand, that loads the coverage results during execute. And all reports overrides the execute and do something additional.

Reply:
Option in this situation has a value. The command shouldn't know about option initialization.
I can move logic from CoverageFileOption into this object but I don't think it is a good way.

Reply:
About override value, it is not overridden. We can cast object on different interfaces. Each interface will have different value type.

Reply:
Sorry to say that but I don't like at all property overriding. why not using a ctor for that
```csharp
protected abstract string CommandName { get; }
protected abstract string CommandDescription { get; }
protected abstract IEnumerable<IMiniCoverOption> MiniCoverOptions { get; }
```

would be better with:
```csharp
protected string CommandName { get; }
protected string CommandDescription { get; }
protected IEnumerable<IMiniCoverOption> MiniCoverOptions { get; }

protected BaseCommand(string commandName, string descriptions, params IMiniCoverOption[] options)
```

Reply:
Comment below